corehq/ex-submodules/casexml/apps/phone/xml.py

```diff
@@ -129,10 +130,25 @@ def get_registration_element_data(restore_user):
         "password": restore_user.password,
         "uuid": restore_user.user_id,
         "date": date_to_xml_string(restore_user.date_joined),
-        "user_data": restore_user.user_session_data
+        "user_data": get_user_data_for_restore(restore_user)
     }
 
 
+def get_user_data_for_restore(restore_user):
+    user_data = restore_user.user_session_data
+    for custom_user_data in get_custom_user_data_for_restore(restore_user):
+        user_data.update(custom_user_data)
```

Comment:
Should you consider `copy` here since this might update the passed value?

Reply:
Think copy is not needed as this `custom_user_data` object instance won't be used anywhere else.